# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1453399407 18000
#      Thu Jan 21 13:03:27 2016 -0500
# Node ID 61ca546e19428639520724a70cf648a1ad855a2b
# Parent  df789122470905908f1f549fda31dff450755146
improve handling of --disable-docs

* doc/module.mk (INFO_DEPS, DVIS, PDFS, PSS, HTMLS):
Define conditionally.

diff --git a/doc/module.mk b/doc/module.mk
--- a/doc/module.mk
+++ b/doc/module.mk
@@ -23,16 +23,26 @@ am__v_TEX_ = $(am__v_TEX_$(AM_DEFAULT_VE
 am__v_TEX_0 = @echo "  TEX     " $@;
 am__v_TEX_1 =
 
 AM_V_PDFTEX = $(am__v_PDFTEX_$(V))
 am__v_PDFTEX_ = $(am__v_PDFTEX_$(AM_DEFAULT_VERBOSITY))
 am__v_PDFTEX_0 = @echo "  PDFTEX  " $@;
 am__v_PDFTEX_1 =
 
+## Conditionally define the following variables so that --disable-docs
+## can work.  If we don't define them, Automake will always define them,
+## even when AMCOND_BUILD_DOCS is false.
+
+INFO_DEPS =
+DVIS =
+PDFS =
+PSS =
+HTMLS =
+
 if AMCOND_BUILD_DOCS
 
 ## Listing "dvi" explicitly here seems to cause two simultaneous
 ## builds of the DVI files, presumably because the PS version
 ## also depends on the DVI file and somehow the rules are invoked
 ## twice.  Is that a bug in automake or make or what?
 
 doc_EXTRA_DIST += doc/texmf.cnf
@@ -230,16 +240,22 @@ BUILT_OCTAVE_TEXI_SRC = \
   $(MUNGED_TEXI_SRC)
 
 info_TEXINFOS += \
   doc/interpreter/octave.texi
 
 octave_TEXINFOS = \
   $(BUILT_OCTAVE_TEXI_SRC)
 
+INFO_DEPS += $(srcdir)/doc/interpreter/octave.info
+DVIS += doc/interpreter/octave.dvi
+PDFS += doc/interpreter/octave.pdf
+PSS += doc/interpreter/octave.ps
+HTMLS += doc/interpreter/octave.html
+
 ## As of version 1.14.1, automake does not seem to generate
 ## rules for DVI, PDF, or HTML output that work for us when
 ## there are additional dependencies, so we include our own
 ## versions of the rules here.
 
 OCTAVE_HTML_DIR = doc/interpreter/octave.html
 OCTAVE_HTML_TMP_DIR = $(OCTAVE_HTML_DIR:.html=.htp)
 OCTAVE_HTML_STAMP = $(OCTAVE_HTML_DIR)/.octave-html-stamp
@@ -428,16 +444,22 @@ liboctave_TEXINFOS = \
   doc/liboctave/optim.texi \
   doc/liboctave/preface.texi \
   doc/liboctave/quad.texi \
   doc/liboctave/range.texi
 
 info_TEXINFOS += \
   doc/liboctave/liboctave.texi
 
+INFO_DEPS += $(srcdir)/doc/liboctave/liboctave.info
+DVIS += doc/liboctave/liboctave.dvi
+PDFS += doc/liboctave/liboctave.pdf
+PSS += doc/liboctave/liboctave.ps
+HTMLS += doc/liboctave/liboctave.html
+
 doc/liboctave/liboctave.dvi: doc/liboctave/liboctave.texi $(srcdir)/doc/liboctave/version-liboctave.texi | doc/liboctave/$(am__dirstamp)
 	$(AM_V_TEXI2DVI)TEXINPUTS="$(am__TEXINFO_TEX_DIR)$(PATH_SEPARATOR)$$TEXINPUTS" \
 	MAKEINFO='$(MAKEINFO) $(AM_MAKEINFOFLAGS) $(MAKEINFOFLAGS) -I doc/liboctave -I $(srcdir)/doc/liboctave' \
 	$(TEXI2DVI) $(AM_V_texinfo) --build-dir=$(@:.dvi=.t2d) -o $@ $(AM_V_texidevnull) \
 	`test -f 'doc/liboctave/liboctave.texi' || echo '$(srcdir)/'`doc/liboctave/liboctave.texi
 
 doc/liboctave/liboctave.pdf: doc/liboctave/liboctave.texi $(srcdir)/doc/liboctave/version-liboctave.texi | doc/liboctave/$(am__dirstamp)
 	$(AM_V_TEXI2PDF)TEXINPUTS="$(am__TEXINFO_TEX_DIR)$(PATH_SEPARATOR)$$TEXINPUTS" \
